baremetal_network_provisioning/tests/unit/ml2/test_mechanism_hp.py

```diff
@@ -0,0 +1,189 @@
+            vnic_type.assert_called_with(fake_context)
+            d_port.assert_called_with(fake_port_id)'''
+
+    def test_update_port_precommit(self):
+        """Test update_port_precommit method."""
+        fake_port_dict = mock.Mock()
+        fake_context = mock.Mock()
+        with contextlib.nested(
+            mock.patch.object(hp_mech.HPMechanismDriver,
+                              '_is_port_of_interest',
+                              return_value=True),
+            mock.patch.object(hp_mech.HPMechanismDriver,
+                              '_construct_port',
+                              return_value=fake_port_dict),
+            mock.patch.object(np_drv.HPNetworkProvisioningDriver,
+                              'update_port',
+                              return_value=None)
+        ) as (is_port, cons_port, u_port):
+            self.driver.update_port_precommit(fake_context)
+            is_port.assert_called_with(fake_context)
+            cons_port.assert_called_with(fake_context, False)
+            u_port.assert_called_with(fake_port_dict)
+
+    def test__construct_port(self):
+        """Test _construct_port method."""
+        tenant_id = 'ten-1'
+        network_id = 'net1-id'
+        segmentation_id = 1001
+        vm_id = 'vm1'
+        fake_port_dict = self._get_port_dict()
+        network_context = self._get_network_context(tenant_id,
+                                                    network_id,
+                                                    segmentation_id,
+                                                    False)
+
+        port_context = self._get_port_context(tenant_id,
+                                              network_id,
+                                              vm_id,
+                                              network_context)
+        port_dict = self.driver._construct_port(port_context, segmentation_id)
+        self.assertEqual(port_dict, fake_port_dict)
+
+
+class FakeNetworkContext(object):
+    """To generate network context for testing purposes only."""
+
+    def __init__(self, network, segments=None, original_network=None):
+        self._network = network
+        self._original_network = original_network
+        self._segments = segments
+
+    @property
+    def current(self):
+        return self._network
+
+    @property
+    def original(self):
+        return self._original_network
+
+    @property
+    def network_segments(self):
+        return self._segments
+
+
+class FakePortContext(object):
+    """To generate port context for testing purposes only."""
+
+    def __init__(self, port, original_port, network):
+        self._port = port
+        self._original_port = original_port
+        self._network_context = network
+
+    @property
+    def current(self):
+        return self._port
+
+    @property
+    def original(self):
+        return self._original_port
+
+    @property
+    def network(self):
+        return self._network_context
```

Comment:
test_delete_port_precommit test case is there but in commented state. once the portbindings is updated with VNIC_TYPE, I will uncomment it.